test/common/3Scale_user_promotion.go

```diff
@@ -0,0 +1,93 @@
+import (
+	goctx "context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/test/resources"
+	"k8s.io/apimachinery/pkg/util/wait"
+	runtimeConfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+	"testing"
+	"time"
+)
+
+func Test3ScaleUserPromotion(t *testing.T, ctx *TestingContext) {
+
+	var (
+		developerUser      = fmt.Sprintf("%v-%d", DefaultTestUserName, 0)
+		dedicatedAdminUser = fmt.Sprintf("%v-%d", defaultDedicatedAdminName, 0)
+	)
+
+	if err := createTestingIDP(t, goctx.TODO(), ctx.Client, ctx.KubeConfig, ctx.SelfSignedCerts); err != nil {
+		t.Fatalf("error while creating testing idp: %v", err)
+	}
+
+	// get console master url
+	rhmi, err := getRHMI(ctx.Client)
+	if err != nil {
+		t.Fatalf("error getting RHMI CR: %v", err)
+	}
+
+	// Get the 3Scale host url from the rhmi status
+	host := rhmi.Status.Stages[v1alpha1.ProductsStage].Products[v1alpha1.Product3Scale].Host
+	keycloakHost := rhmi.Status.Stages[v1alpha1.AuthenticationStage].Products[v1alpha1.ProductRHSSO].Host
+	redirectUrl := fmt.Sprintf("%v/p/admin/dashboard", host)
+
+	loginTo3ScaleAsDevloper(t, developerUser, host)
+
+	err = loginToThreeScale(t, host, dedicatedAdminUser, DefaultPassword, "testing-idp", ctx.HttpClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tsClient := resources.NewThreeScaleAPIClient(host, keycloakHost, redirectUrl, ctx.HttpClient, ctx.Client, t)
+
+	// Make sure 3Scale is available
+	err = tsClient.Ping()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	userId, err := tsClient.GetUserId(developerUser)
+	if err != nil || userId == "" {
+		t.Fatal("Failed to retrieve user id for ", developerUser, "userId: ", userId, err)
+	}
+
+	err = tsClient.SetUserAsAdmin(developerUser, fmt.Sprintf("%v@example.com", developerUser), userId)
+	if err != nil {
+		t.Fatal("Failed to set user as admin ", err)
+	}
+
+	// TODO: Waiting an arbitrary amount of time to verify that a 3Scale reconcile was complete
+	// and did not result in reverting the promotion of test-user to admin
+	// Change this when https://issues.redhat.com/browse/INTLY-7770 implemented
+	wait.Poll(time.Second*350, time.Minute*7, func() (done bool, err error) {
+
+		isAdmin, err := tsClient.VerifyUserIsAdmin(userId)
+		if err != nil {
+			t.Fatal("Error attempting to verify that the user is an admin ", err)
+		}
+		if !isAdmin {
+			t.Fatal("User reverted from admin back to member")
+		}
+
+		return true, nil
+	})
+
+}
+
+// Login as a developer and create a separate testing context. This will provide a separate http client, mimicking
+// What would happen in reality, i.e. 2 users using separate browsers.
+func loginTo3ScaleAsDevloper(t *testing.T, user string, host string) error {
+	config, err := runtimeConfig.GetConfig()
+	if err != nil {
+		t.Fatalf("Failed to get runtime config: %v", err)
+	}
+	ctx, err := NewTestingContext(config)
+
+	err = loginToThreeScale(t, host, user, DefaultPassword, "testing-idp", ctx.HttpClient)
```

Comment:
This might give a `kubeconfig refined` error due to https://github.com/operator-framework/operator-sdk/issues/2167 :thinking: 

If you just need a new http client, perhaps you could use this function instead of creating a new testing context :thinking: :
https://github.com/integr8ly/integreatly-operator/blob/955d4997f97944aed60aef1c6184c5e9e4e8f0ef/test/common/shared_functions.go#L170

```suggestion
func loginTo3ScaleAsDevloper(t *testing.T, user string, host string, ctx *TestingContext) error {
	httpClient, err := NewTestingHTTPClient(ctx.KubeConfig)
	
	if err != nil {
	  t.Fatal(err)
	}

	err = loginToThreeScale(t, host, user, DefaultPassword, "testing-idp", httpClient)
```

Reply:
Thanks @KevFan Was not aware of that